bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/WDChannels.java

```diff
@@ -44,4 +59,117 @@ public synchronized WorkerDataChannel getWorkerDataChannel(String name) {
         }
         return channel;
     }
+
+    public Object receiveDataMultipleChannels(Strand strand, ReceiveField[] receiveFields, Type targetType)
+            throws Throwable {
+        if (strand.workerReceiveMap == null) {
+            strand.workerReceiveMap = ValueCreator.createMapValue(targetType);
+        }
+        for (ReceiveField field : receiveFields) {
+            WorkerDataChannel channel = getWorkerDataChannel(field.channelName());
+            if (!channel.isClosed()) {
+                Object result = channel.tryTakeData(strand, true);
+                checkAndPopulateResult(strand, field, result, channel);
+            } else {
+                if (channel.getState() == WorkerDataChannel.State.AUTO_CLOSED) {
+                    checkAndPopulateResult(strand, field, ErrorUtils.createNoMessageError(field.channelName()),
+                            channel);
+                }
+            }
+        }
+        return clearResultCache(strand, receiveFields);
+    }
+
+    private void checkAndPopulateResult(Strand strand, ReceiveField field, Object result, WorkerDataChannel channel) {
+        if (result != null) {
```

Comment:
can invert the condition and exit early. this would make code more readable by reducing the level of indentation